test/built-ins/object/entries/getter-adding-key.js

```diff
@@ -0,0 +1,32 @@
+// Copyright (C) 2015 Jordan Harband. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: Object.entries sees a new element added by a getter that is hit during iteration
+es7id: pending
+author: Jordan Harband
+---*/
+
+var bAddsC = {
+    a: 'A',
+    get b() {
+        this.c = 'C';
+        return 'B';
+    }
+};
+
+var result = Object.entries(bAddsC);
+
+assert.sameValue(Array.isArray(result), true, 'result is an array');
+assert.sameValue(result.length, 3, 'result has 3 items');
```

Comment:
That's not how the spec language is written - it explicitly allows for that, just like the algorithm for `Reflect.enumerate`.